sklearnex/preview/linear_model/linear.py

```diff
@@ -300,9 +300,13 @@ def _onedal_fit(self, X, y, sample_weight, queue=None):
                 )
 
             self._initialize_onedal_estimator()
-            self._onedal_estimator.fit(X, y, queue=queue)
+            try:
+                self._onedal_estimator.fit(X, y, queue=queue)
 
-            self._save_attributes()
+                self._save_attributes()
+            except RuntimeError:
+                del self._onedal_estimator
+                super().fit(X, y)

```

Comment:
Looks weird. Yes, it will fix the problem for now. Unfortunately we don't quite understand what is causing the problem on onedal side